quiz/01-rate-limit/quiz01/core/tests.py

```diff
@@ -0,0 +1,47 @@
+from django.test import TestCase
+from django.test import client
+import time
+# Create your tests here.
+
+class APITestCase(TestCase):
+
+    def setUp(self):
+        self.c = client.Client()
+    
+    # Unover rate limiting get test 
+    # will be get 200 status
+    def test_get_not_over(self):
+        time.sleep(2)
+        print('-----------Get not over-----------')
+        for i in range(1,101):
+            resp = self.c.get('/api/')
```

Comment:
how do you guarantee these 100 API calls would be executed in one second?

Reply:
In normal circumstances, the testing time will not exceed 1 second, so no special handling is required. In the new version, a judgment of execution time has been added. When the execution time exceeds 1 second, the test will be performed again. If the execution time is less than 1 second or has been tested 5 times (if it cannot be less than 1 second, an error will be displayed), the testing loop will be break or terminated.
```
for i in range(1,5):  
    t0 = int(time.time())

    // request  //

    t1 = int(time.time())
    if((t1-t0) <1):
        break
```

Reply:
this would trigger infinite testing when the worse scenario has happened. why not just fail the test case?